packages/python/accern_data/accern_data.py

```diff
@@ -245,34 +265,66 @@ def max_date(
         # Second last date.
         return max(dates[:-1]).strftime(DT_FORMAT)
 
-    def do_init(self, is_first_day: bool) -> None:
+    def do_init(self, indicator: ProgressIndicator) -> None:
         fname = self.get_path(self._is_by_day)
-        if is_first_day or self._is_by_day:
+        if self._begining_of_file or self._is_by_day:
             pd.DataFrame([], columns=self._cols).to_csv(
                 fname, index=False, header=True, mode="w")
-        print_fn(f"current file is {fname}")
+        indicator.log(f"current file is {fname}")
 
-    def add_result(
-            self,
-            signal: Union[pd.DataFrame, List[Dict[str, Any]]]) -> None:
+    def add_result(self, signal: pd.DataFrame) -> None:
         fname = self.get_path(is_by_day=self._is_by_day)
-        assert isinstance(signal, pd.DataFrame)
         signal.to_csv(fname, index=False, header=False, mode="a")
 
-    def finish_day(self) -> None:
+    def finish_day(self, indicator: ProgressIndicator) -> None:
         # csv files are saved by add_result
-        pass
+        self._begining_of_file = self._is_by_day
```

Comment:
typo

Reply:
what do you need _beginning_of_file for anyway? just use _cols and whether it is None